### ISSUE TITLE
<amp-app-banner> is broken

### ISSUE THREAD

**Issue by nainar** [ampproject/amphtml#27065]
https://amp.dev/documentation/examples/components/amp-app-banner/?format=websites shows it to be broken. 

In this case it is a 404 on the .json but even examples/article.html which has this banner isn't working. 

@ampproject/wg-ui-and-a11y @caroqliu

**Comment by wassgha** [comment]
@caroqliu Naina and I investigated this and it seems like it has something to do with `layoutCallback` not being called since the `amp-app-banner` has a `nodisplay` layout (although most of the work happens inside `layoutCallback`).

**Comment by caroqliu** [comment]
@sebastianbenz and I discussed this issue further, two important things to note here:

(1) The manifest that informs the banner on the amp.dev sample was missing and has been [reintroduced](https://github.com/ampproject/amp.dev/pull/3642)
(2) By 1, the amp.dev sample linked in the original issue is now working on Chrome and Safari on iOS both [in and out of the AMP viewer](https://amp.dev/documentation/components/amp-app-banner/#os/browser/viewer-dependencies). 

@nainar Can you provide more context on the environments in which the component was originally broken? Want to make sure we cover all our bases here. Because the banner's function is to link to native app downloads, I'd like to confirm the expected behavior for desktop contexts--I assume it is to be ineffective, explaining the `nodisplay` behavior that @wassgha brought up.

**Comment by nainar** [comment]
Yeah in the desktop case it is not laying out - unless you have `#webview=1` as shown here: https://amp.dev/documentation/examples/components/amp-app-banner/?format=websites#demo
